code/modules/modular_computers/computers/modular_computer/hardware.dm

```diff
@@ -55,6 +55,17 @@
 			return
 		found = TRUE
 		tesla_link = H
+	else if(istype(H, /obj/item/device/paicard))
+		if(personal_ai)
+			to_chat(user, SPAN_WARNING("\The [src]'s personal AI slot is already occupied by \the [personal_ai]."))
+			return
+		personal_ai = H
+		to_chat(user, SPAN_NOTICE("You install \the [H] into \the [src]."))
+		personal_ai.pai.parent_computer = src
+		personal_ai.pai.verbs += /mob/proc/personal_computer_interact
+		to_chat(personal_ai.pai, SPAN_NOTICE("You gain access to \the [src]'s computronics."))
+		user.drop_from_inventory(H, src)
+		update_icon()
```

Comment:
Two things:

1) This sounds like you want to make a proc on the PC to eject the pai
2) I'm pretty sure `\The [src]'s personal AI slot is already occupied by \the [personal_ai].` prints `"The modular computer's personal AI slot is already occupied by the BUDDY"` or whatever the pAI name is, just say it's already occupied period

Reply:
since the pAI's name is a proper name, it won't include "the", i believe. and no, this is what i want, it works